web/src/context/auth-provider.tsx

```diff
@@ -0,0 +1,155 @@
+import {useEffect, createContext, useState, useContext} from 'react';
+
+export interface User {
+  user: {
+    id: string;
+    name: string;
+    email: string;
+  };
+  token: string;
+  refreshToken: string;
+}
+
+export interface AuthContext {
+  isAuthenticated: boolean;
+  loginWithToken: (
+    token?: string,
+    refreshToken?: string
+  ) => Promise<{
+    status: string;
+    message: string;
+  }>;
+  logout: () => void;
+  user: User | null;
+}
+
+const AuthContext = createContext<AuthContext | null>(null);
+
+const key = 'user';
+
+/**
+ * Stores or removes the user object in/from localStorage.
+ * @param {User | null} user - The user object to store or `null` to remove the user.
+ */
+function setStoredUser(user: User | null) {
+  if (user) {
+    localStorage.setItem(key, JSON.stringify(user));
+  } else {
+    localStorage.removeItem(key);
+  }
+}
+
+/**
+ * Provides authentication context to its children.
+ * @param {{children: React.ReactNode}} props - The children components that will receive the auth context.
+ * @returns {JSX.Element} The AuthProvider component.
+ */
+export function AuthProvider({children}: {children: React.ReactNode}) {
+  const [user, setUser] = useState<User | null>(
+    JSON.parse(localStorage.getItem(key) || 'null')
+  );
+
+  const isAuthenticated = !!user;
+
+  /**
+   * Logs out the user by removing the stored user object and setting the user to null.
+   */
+  const logout = () => {
+    setStoredUser(null);
+    setUser(null);
+  };
+
+  /**
+   * Logs in the user with a token and refresh token.
```

Comment:
This is kind of a weird phrasing of this - you are not logging in the user right? You are just retrieving fresh user details based on the token (which implies the user was already logged in?)